app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.campusguide.map.infoWindow
+
+import CustomInfoWindow
+import android.content.Context
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.gms.maps.model.Polygon
+
+class BuildingClickListener(private val context: Context, private val googleMap: GoogleMap) : GoogleMap.OnPolygonClickListener{
+
+    fun buildingInfoWindow(location: LatLng, info: InfoWindowData) {
+        val customInfoWindow = CustomInfoWindow(context)
+        googleMap.setInfoWindowAdapter(customInfoWindow)
+
+        // Clears any existing markers from the GoogleMap
+        //googleMap.clear()
+
+        // Creating an instance of MarkerOptions to set position
+        val markerOptions = MarkerOptions()
+
+        // Setting position on the MarkerOptions
+        markerOptions.position(location)
+
+        //Set Custom InfoWindow Adapter
+        googleMap.setInfoWindowAdapter(customInfoWindow)
+
+        // Animating to the currently touched position
+        googleMap.animateCamera(CameraUpdateFactory.newLatLng(location))
```

Comment:
This makes the app center on the marker, cutting off the top half of the window.  Is there any way to center the camera on the widow instead? It should be a constant offset